vagabond/gui/AddEntity.cpp

```diff
@@ -352,10 +366,24 @@ void AddEntity::buttonPressed(std::string tag, Button *button)
 	}
 	else if (tag == "refine")
 	{
-		SerialRefiner *refiner = new SerialRefiner(this, &_obj);
-		refiner->setJobType(rope::ThoroughRefine);
-		refiner->show();
-		return;
+                try
+                {
+                        if (_obj.modelCount()==0)
+                        {
+                                throw std::invalid_argument("No models detected, have you modeled your sequences?");
+                        }
+                        SerialRefiner *refiner = new SerialRefiner(this, &_obj);
+                        refiner->setJobType(rope::ThoroughRefine);
+                        refiner->show();
+                        return;
+
+                }
+                catch(std::invalid_argument &err)
```

Comment:
Since there is only one way of handling these exceptions (taking the `.what()` and putting it into a `BadChoice` alert), there will be a way of designing the code so there is only ever one `catch` statement to handle everything. Duplicated code is a form of [technical debt](https://en.wikipedia.org/wiki/Technical_debt).